packages/vaadin-virtual-list/src/virtualizer-iron-list-adapter.js

```diff
@@ -0,0 +1,432 @@
+    }
+
+    this.__skipNextVirtualIndexAdjust = true;
+    super.scrollToIndex(targetVirtualIndex);
+    if (this.firstVisibleIndex !== index - this._vidxOffset && this._scrollTop < this._maxScrollTop && !this.grid) {
+      // Second invocation to scrollToIndex may be needed to workaround an issue in iron-list
+      super.scrollToIndex(targetVirtualIndex);
+    }
+    this._scrollHandler();
+  }
+
+  flush() {
+    this._resizeHandler();
+    flush();
+    this._scrollHandler();
+    this.__scrollReorderDebouncer && this.__scrollReorderDebouncer.flush();
+    this.__debouncerWheelAnimationFrame && this.__debouncerWheelAnimationFrame.flush();
+  }
+
+  update(startIndex = 0, endIndex = this.size - 1) {
+    this.__getVisibleElements().forEach((el) => {
+      if (el.__virtualIndex >= startIndex && el.__virtualIndex <= endIndex) {
+        this.updateElement(el, el.__virtualIndex);
+      }
+    });
+  }
+
+  __getIndexScrollOffset(index) {
+    const element = this.__getVisibleElements().find((el) => el.__virtualIndex === index);
+    return element ? this.scrollTarget.getBoundingClientRect().top - element.getBoundingClientRect().top : undefined;
+  }
+
+  set size(size) {
+    if (size === this.size) {
+      return;
+    }
+
+    let fvi = this.firstVisibleIndex + this._vidxOffset;
+
+    const fviOffsetBefore = this.__getIndexScrollOffset(fvi);
+
+    this.__size = size;
+    this._itemsChanged({
+      path: 'items'
+    });
+
+    fvi = Math.min(fvi, size - 1);
+    this.scrollToIndex(fvi);
+
+    const fviOffsetAfter = this.__getIndexScrollOffset(fvi);
+    if (fviOffsetBefore !== undefined && fviOffsetAfter !== undefined) {
+      this._scrollTop += fviOffsetBefore - fviOffsetAfter;
+    }
+
+    if (!this.elementsContainer.children.length) {
+      requestAnimationFrame(() => this._resizeHandler());
+    }
+  }
+
+  get size() {
+    return this.__size;
+  }
+
+  /** @private */
+  get _scrollTop() {
+    return this.scrollTarget.scrollTop;
+  }
+
+  /** @private */
+  set _scrollTop(top) {
+    this.scrollTarget.scrollTop = top;
+  }
+
+  /** @private */
+  get items() {
+    return {
+      length: Math.min(this.size, MAX_VIRTUAL_COUNT)
+    };
+  }
+
+  /** @private */
+  get offsetHeight() {
+    return this.scrollTarget.offsetHeight;
+  }
+
+  /** @private */
+  get $() {
+    return {
+      items: this.scrollContainer
+    };
+  }
+
+  /** @private */
+  updateViewportBoundaries() {
+    const styles = window.getComputedStyle(this.scrollTarget);
+    this._scrollerPaddingTop = this.scrollTarget === this ? 0 : parseInt(styles['padding-top'], 10);
+    this._isRTL = Boolean(styles.direction === 'rtl');
+    this._viewportWidth = this.elementsContainer.offsetWidth;
+    this._viewportHeight = this.scrollTarget.offsetHeight;
+    this._scrollPageHeight = this._viewportHeight - this._scrollLineHeight;
+    this.grid && this._updateGridMetrics();
+  }
+
+  /** @private */
+  setAttribute() {}
```

Comment:
The `iron-list.js` upgrade [added a call](https://github.com/vaadin/web-components/pull/321/files#diff-f57f57e2330884cd983e6e090ff2d56668cc30052baa175f70e9d888d7598713R737) to `setAttribute` to add an attribute to the iron-list host element itself. Since the adapter isn't an HTMLElement, we need to mock the function to avoid an error (or having to modify iron-list source code).

If the attribute is needed on the host element of `<vaadin-virtual-list>`, it's possible to add a `hostElement` property to the configuration in scope of the `virtual-list` implementation.